File: exercises/beecrowd/1534/1534.ts
// https://judge.beecrowd.com/pt/problems/view/1534
import * as fs from "fs";

function main(): void {
  const input = fs
    .readFileSync("/dev/stdin", "utf8")
    .trim()
    .split("\n")
    .map(Number);
  let lineIndex = 0;

  while (lineIndex < input.length) {
    const n = input[lineIndex++];

    if (isNaN(n)) {
      break;
    }

    const matrix: number[][] = [];
    for (let i = 0; i < n; i++) {
      matrix[i] = [];
      for (let j = 0; j < n; j++) {
        if (i + j === n - 1) {
          matrix[i][j] = 2;
        } else if (i === j) {
          matrix[i][j] = 1;
        } else {
          matrix[i][j] = 3;
        }
      }
    }

    for (let i = 0; i < n; i++) {
      let rowString = "";
      for (let j = 0; j < n; j++) {
        rowString += String(matrix[i][j]);
      }
      console.log(rowString);
    }
  }
}

main();

File: exercises/beecrowd/1534/1534.go
// https://judge.beecrowd.com/pt/problems/view/1534
package main

import (
	"fmt"
)

func main() {
	var n int
	for {
		_, err := fmt.Scan(&n)
		if err != nil {
			break
		}

		matrix := make([][]int, n)
		for i := 0; i < n; i++ {
			matrix[i] = make([]int, n)
			for j := 0; j < n; j++ {
				if i+j == n-1 {
					matrix[i][j] = 2
				} else if i == j {
					matrix[i][j] = 1
				} else {
					matrix[i][j] = 3
				}
			}
		}

		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				fmt.Printf("%d", matrix[i][j])
			}
			fmt.Println()
		}
	}
}
